### PR TITLE
Eip 30

### PR DESCRIPTION
**Pull request by carlospzurita**
Added a dynamic dependency tree on the ExecutableTestSuiteView to control the activated Test Suites and their dependencies.

### CODE REVIEW

**Comment by carlospzurita** [comment]
We reviewed the Codacy reports and will provide a new commit fixing the issues reported on this PR

File: src/main/webapp/js/app/views/ExecutableTestSuiteView.js [hunk 1]
```diff
@@ -27,10 +27,57 @@ define([
...
+        let classList = document.querySelector("option[value="+id+"]").parentElement.parentElement.classList;
+        if (typeof model.attributes.dependencies !== "undefined" && !classList.contains("ui-disabled")){
+            var idDependency = model.attributes.dependencies.executableTestSuite.ref;
+            if(!dependenciesTree[idDependency]){
+                dependenciesTree[idDependency] = [];
+            }
+            let classListDependency = document.querySelector("option[value="+idDependency+"]").parentElement.parentElement.classList;
+            let activating = classList.contains("ui-flipswitch-active");
+            if(activating){
+                dependenciesTree[idDependency].push(id);
```
[the remark this answers is not in the document]
Comment (carlospzurita) [reply src/main/webapp/js/app/views/ExecutableTestSuiteView.js]: This issue is only relevant if the variable used as a key comes from an user input, as they can inject malicious code to be validated on the server. In this case, idDependency is always calculated from the DOM, and no information is sent to the server, so we think that this is not a security issue.

File: src/main/webapp/js/app/views/ExecutableTestSuiteView.js [hunk 2]
```diff
@@ -27,10 +27,57 @@ define([
...
+            }else{
+                dependenciesTree[idDependency].splice(dependenciesTree[idDependency].indexOf(id),1);
```
[the remark this answers is not in the document]
Comment (carlospzurita) [reply src/main/webapp/js/app/views/ExecutableTestSuiteView.js]: This issue is only relevant if the variable used as a key comes from an user input, as they can inject malicious code to be validated on the server. In this case, idDependency is always calculated from the DOM, and no information is sent to the server, so we think that this is not a security issue.

File: src/main/webapp/js/app/views/ExecutableTestSuiteView.js [hunk 3]
```diff
@@ -27,10 +27,57 @@ define([
...
+            }
+            if(activating && !classListDependency.contains("ui-flipswitch-active")){
+                classListDependency.add("ui-flipswitch-active");
+                markDependants(idDependency);
+            }
+            if(activating && !classListDependency.contains("ui-disabled")){
+                classListDependency.add("ui-disabled");
+            }
+            if(dependenciesTree[idDependency].length === 0){
```
[the remark this answers is not in the document]
Comment (carlospzurita) [reply src/main/webapp/js/app/views/ExecutableTestSuiteView.js]: This issue is only relevant if the variable used as a key comes from an user input, as they can inject malicious code to be validated on the server. In this case, idDependency is always calculated from the DOM, and no information is sent to the server, so we think that this is not a security issue.

File: src/main/webapp/js/app/views/ExecutableTestSuiteView.js [hunk 4]
```diff
@@ -27,10 +27,57 @@ define([
...
+	var dependenciesTree = {};
+	
+	//Function to recursively switch all the dependents tests on selecting a test
+    function markDependants(id){
+        var model = router.executableTestSuitesView.collection.models.filter((x) => x.id === id)[0];
+        let classList = document.querySelector("option[value="+id+"]").parentElement.parentElement.classList;
+        if (typeof model.attributes.dependencies !== "undefined" && !classList.contains("ui-disabled")){
+            var idDependency = model.attributes.dependencies.executableTestSuite.ref;
+            if(!dependenciesTree[idDependency]){
+                dependenciesTree[idDependency] = [];
```
[the remark this answers is not in the document]
Comment (carlospzurita) [reply src/main/webapp/js/app/views/ExecutableTestSuiteView.js]: This issue is only relevant if the variable used as a key comes from an user input, as they can inject malicious code to be validated on the server. In this case, idDependency is always calculated from the DOM, and no information is sent to the server, so we think that this is not a security issue.

File: src/main/webapp/js/app/views/ExecutableTestSuiteView.js [hunk 5]
```diff
@@ -27,10 +27,57 @@ define([
...
+	
+	var dependenciesTree = {};
+	
+	//Function to recursively switch all the dependents tests on selecting a test
+    function markDependants(id){
+        var model = router.executableTestSuitesView.collection.models.filter((x) => x.id === id)[0];
+        let classList = document.querySelector("option[value="+id+"]").parentElement.parentElement.classList;
+        if (typeof model.attributes.dependencies !== "undefined" && !classList.contains("ui-disabled")){
+            var idDependency = model.attributes.dependencies.executableTestSuite.ref;
+            if(!dependenciesTree[idDependency]){
```
[the remark this answers is not in the document]
Comment (carlospzurita) [reply src/main/webapp/js/app/views/ExecutableTestSuiteView.js]: This issue is only relevant if the variable used as a key comes from an user input, as they can inject malicious code to be validated on the server. In this case, idDependency is always calculated from the DOM, and no information is sent to the server, so we think that this is not a security issue.

File: src/main/webapp/js/app/views/ExecutableTestSuiteView.js [hunk 6]
```diff
@@ -27,10 +27,57 @@ define([
     "toastr",
     "../models/ExecutableTestSuiteModel",
 ], function( $, Backbone, v2, EtfView, moment, ExecutableTestSuiteModel ) {
-
+	
+	var dependenciesTree = {};
+	
+	//Function to recursively switch all the dependents tests on selecting a test
+    function markDependants(id){
+        var model = router.executableTestSuitesView.collection.models.filter((x) => x.id === id)[0];
```
[the remark this answers is not in the document]
Comment (carlospzurita) [reply src/main/webapp/js/app/views/ExecutableTestSuiteView.js]: The reference to `router` is added on compilation time.

File: src/main/webapp/js/app/views/ExecutableTestSuiteView.js [hunk 7]
```diff
@@ -27,13 +27,70 @@ define([
...
+	//Function to recursively switch all the dependents tests on selecting a test
+	//CODACY bot: marked as too complex. Would need to refactor, but functionality works fine
+    function markDependants(id){
+    	//Router instanced in compilation time
+        var model = router.executableTestSuitesView.collection.models.filter((x) => x.id === id)[0];
+        let classList = document.querySelector("option[value="+id+"]").parentElement.parentElement.classList;
+        if (typeof model.attributes.dependencies !== "undefined" && !classList.contains("ui-disabled")){
+            var idDependency = model.attributes.dependencies.executableTestSuite.ref;
+            //CODACY bot: Marked as object injection. Dependencies tree has no sensible information and is not used in any CRUD operation
+            if(!dependenciesTree[idDependency]){
```
Comment (jonherrmann) [inline src/main/webapp/js/app/views/ExecutableTestSuiteView.js]: Codacy Issue found: [Generic Object Injection Sink (security/detect-object-injection)](https://app.codacy.com/manual/interactive-instruments/etf-webapp/pullRequest?prid=3592488)

**Comment by jonherrmann** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 9
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/interactive-instruments/etf-webapp/pullRequest?prid=3592488&bid=12749860)

File: src/main/webapp/js/app/views/ExecutableTestSuiteView.js [hunk 8]
```diff
@@ -27,13 +27,70 @@ define([
...
+                //CODACY bot: Marked as object injection. Dependencies tree has no sensible information and is not used in any CRUD operation
+                dependenciesTree[idDependency] = [];
+            }
+            let classListDependency = document.querySelector("option[value="+idDependency+"]").parentElement.parentElement.classList;
+            let activating = classList.contains("ui-flipswitch-active");
+            if(activating){
+            	//CODACY bot: Marked as object injection. Dependencies tree has no sensible information and is not used in any CRUD operation
+                dependenciesTree[idDependency].push(id);
+            }else{
+            	//CODACY bot: Marked as object injection. Dependencies tree has no sensible information and is not used in any CRUD operation
```
Comment (jonherrmann) [inline src/main/webapp/js/app/views/ExecutableTestSuiteView.js]: Codacy Issue found: [Mixed spaces and tabs.](https://app.codacy.com/manual/interactive-instruments/etf-webapp/pullRequest?prid=3592488)

File: src/main/webapp/js/app/views/ExecutableTestSuiteView.js [hunk 9]
```diff
@@ -27,13 +27,70 @@ define([
...
+                dependenciesTree[idDependency].splice(dependenciesTree[idDependency].indexOf(id),1);
```
Comment (jonherrmann) [inline src/main/webapp/js/app/views/ExecutableTestSuiteView.js]: Codacy Issue found: [Generic Object Injection Sink (security/detect-object-injection)](https://app.codacy.com/manual/interactive-instruments/etf-webapp/pullRequest?prid=3592488)

File: src/main/webapp/js/app/views/ExecutableTestSuiteView.js [hunk 10]
```diff
@@ -27,13 +27,70 @@ define([
...
 ], function( $, Backbone, v2, EtfView, moment, ExecutableTestSuiteModel ) {
-
+	
+	var dependenciesTree = {};
+	
+	//Function to recursively switch all the dependents tests on selecting a test
+	//CODACY bot: marked as too complex. Would need to refactor, but functionality works fine
+    function markDependants(id){
+    	//Router instanced in compilation time
+        var model = router.executableTestSuitesView.collection.models.filter((x) => x.id === id)[0];
```
Comment (jonherrmann) [inline src/main/webapp/js/app/views/ExecutableTestSuiteView.js]: Codacy Issue found: ['router' is not defined. (no-undef)](https://app.codacy.com/manual/interactive-instruments/etf-webapp/pullRequest?prid=3592488)

File: src/main/webapp/js/app/views/ExecutableTestSuiteView.js [hunk 11]
```diff
@@ -27,13 +27,70 @@ define([
...
+            }
+            if(activating && !classListDependency.contains("ui-flipswitch-active")){
+                classListDependency.add("ui-flipswitch-active");
+                markDependants(idDependency);
+            }
+            if(activating && !classListDependency.contains("ui-disabled")){
+                classListDependency.add("ui-disabled");
+            }
+          //CODACY bot: Marked as object injection. Dependencies tree has no sensible information and is not used in any CRUD operation
+            if(dependenciesTree[idDependency].length === 0){
```
Comment (jonherrmann) [inline src/main/webapp/js/app/views/ExecutableTestSuiteView.js]: Codacy Issue found: [Generic Object Injection Sink (security/detect-object-injection)](https://app.codacy.com/manual/interactive-instruments/etf-webapp/pullRequest?prid=3592488)

File: src/main/webapp/js/app/views/ExecutableTestSuiteView.js [hunk 12]
```diff
@@ -42,43 +99,49 @@ define([
 
             var _this = this;
             this.collection.deferred.done(function() {
+            	//CODACY bot: inherited from master
```
Comment (jonherrmann) [inline src/main/webapp/js/app/views/ExecutableTestSuiteView.js]: Codacy Issue found: [Mixed spaces and tabs.](https://app.codacy.com/manual/interactive-instruments/etf-webapp/pullRequest?prid=3592488)

File: src/main/webapp/js/app/views/ExecutableTestSuiteView.js [hunk 13]
```diff
@@ -27,13 +27,70 @@ define([
...
     "../models/ExecutableTestSuiteModel",
 ], function( $, Backbone, v2, EtfView, moment, ExecutableTestSuiteModel ) {
-
+	
+	var dependenciesTree = {};
+	
+	//Function to recursively switch all the dependents tests on selecting a test
+	//CODACY bot: marked as too complex. Would need to refactor, but functionality works fine
+    function markDependants(id){
+    	//Router instanced in compilation time
```
Comment (jonherrmann) [inline src/main/webapp/js/app/views/ExecutableTestSuiteView.js]: Codacy Issue found: [Mixed spaces and tabs.](https://app.codacy.com/manual/interactive-instruments/etf-webapp/pullRequest?prid=3592488)

File: src/main/webapp/js/app/views/ExecutableTestSuiteView.js [hunk 14]
```diff
@@ -27,13 +27,70 @@ define([
...
+        var model = router.executableTestSuitesView.collection.models.filter((x) => x.id === id)[0];
+        let classList = document.querySelector("option[value="+id+"]").parentElement.parentElement.classList;
+        if (typeof model.attributes.dependencies !== "undefined" && !classList.contains("ui-disabled")){
+            var idDependency = model.attributes.dependencies.executableTestSuite.ref;
+            //CODACY bot: Marked as object injection. Dependencies tree has no sensible information and is not used in any CRUD operation
+            if(!dependenciesTree[idDependency]){
+                //CODACY bot: Marked as object injection. Dependencies tree has no sensible information and is not used in any CRUD operation
+                dependenciesTree[idDependency] = [];
```
Comment (jonherrmann) [inline src/main/webapp/js/app/views/ExecutableTestSuiteView.js]: Codacy Issue found: [Generic Object Injection Sink (security/detect-object-injection)](https://app.codacy.com/manual/interactive-instruments/etf-webapp/pullRequest?prid=3592488)

File: src/main/webapp/js/app/views/ExecutableTestSuiteView.js [hunk 15]
```diff
@@ -27,13 +27,70 @@ define([
...
+            }
+            let classListDependency = document.querySelector("option[value="+idDependency+"]").parentElement.parentElement.classList;
+            let activating = classList.contains("ui-flipswitch-active");
+            if(activating){
+            	//CODACY bot: Marked as object injection. Dependencies tree has no sensible information and is not used in any CRUD operation
+                dependenciesTree[idDependency].push(id);
```
Comment (jonherrmann) [inline src/main/webapp/js/app/views/ExecutableTestSuiteView.js]: Codacy Issue found: [Generic Object Injection Sink (security/detect-object-injection)](https://app.codacy.com/manual/interactive-instruments/etf-webapp/pullRequest?prid=3592488)

**Pull request merged**